Classes/ios/UIViewController+MBDebugPanel.m

```diff
@@ -0,0 +1,157 @@
+{
+    // TODO: test what happens if we call this from viewWillAppear or viewDidDisappear
+
+    if (mb_debugMenuItemsManagementEnabled != self.mb_debugMenuItemsManagementEnabled)
+    {
+        if (mb_debugMenuItemsManagementEnabled == true)
+        {
+            [self swizzleMethod:@selector(viewWillAppear:) withReplacement:JGMethodReplacementProviderBlock {
+                return JGMethodReplacement(void, UIViewController *, BOOL animated) {
+                    JGOriginalImplementation(BOOL, animated);
+                    
+                    [MBDebugPanel addComponentsFromArray:self.mb_managedDebugMenuItems];
+                };
+            }];
+            
+            [self swizzleMethod:@selector(viewDidDisappear:) withReplacement:JGMethodReplacementProviderBlock {
+                return JGMethodReplacement(void, UIViewController *, BOOL animated) {
+                    JGOriginalImplementation(BOOL, animated);
+                    
+                    if ( [MBDebugPanel isPresented] ) {
+                        [MBDebugPanel hide];
+                    }
+                    
+                    [MBDebugPanel removeComponentsInArray:self.mb_managedDebugMenuItems];
+                };
+            }];
+        } else
+        {
+            [self deswizzleMethod:@selector(viewWillAppear:) ];
+            [self deswizzleMethod:@selector(viewDidDisappear:) ];
+        }
+
+        objc_setAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_debugMenuItemsManagementEnabledKey), [NSNumber numberWithBool:mb_debugMenuItemsManagementEnabled], OBJC_ASSOCIATION_RETAIN);
+    }
+}
+
+- (BOOL)mb_debugMenuItemsManagementEnabled
+{
+    BOOL result = NO;
+    
+    id boolObject = objc_getAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_debugMenuItemsManagementEnabledKey) );
+    if ( [boolObject isKindOfClass:[NSNumber class] ] )
+    {
+        NSNumber *boolNumber = boolObject;
+        result = [boolNumber boolValue];
+    }
+    
+    return result;
+}
+
+- (BOOL)viewIsVisible
+{
+    return self.isViewLoaded && self.view.superview != nil;
+}
+
+- (void)mb_addManagedDebugMenuItems:(NSArray *)items
+{
+    if (self.mb_managedDebugMenuItems == nil)
+    {
+        self.mb_managedDebugMenuItems = [NSMutableArray arrayWithArray:items];
+    } else
+    {
+        [self.mb_managedDebugMenuItems addObjectsFromArray:items];
+    }
+    
+    if (self.mb_debugMenuItemsManagementEnabled && self.viewIsVisible)
+    {
+        [MBDebugPanel addComponentsFromArray:items];
+    }
+}
+
+- (void)mb_removeManagedDebugMenuItems:(NSArray *)items
+{
+    [self.mb_managedDebugMenuItems removeObjectsInArray:items];
+}
+
+- (void)setMb_managedDebugMenuItems:(NSMutableArray *)managedDebugMenuItems
+{
+    objc_setAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_managedDebugMenuItemsKey), managedDebugMenuItems, OBJC_ASSOCIATION_RETAIN);
+}
+
+- (NSMutableArray *)mb_managedDebugMenuItems
```

Comment:
See earlier comment about the implementation of `+- (BOOL)mb_debugMenuItemsManagementEnabled`